tasks/Task.cpp

```diff
@@ -0,0 +1,266 @@
+    if(device_info.firmwareVersion.find("1.7") == std::string::npos)
+        RTT::log(RTT::Error) << "Component was developed for firmware version \"1.7\" and actual version is: \""<< device_info.firmwareVersion <<"\". Be aware of eventual incompatibility." << std::endl;
+    RTT::log(RTT::Info) << "USBL's firmware information: Firmware version: \""<< device_info.firmwareVersion <<"\"; Physical and Data-Link layer protocol: \""<<
+            device_info.accousticVersion <<"\"; Manufacturer: \""<< device_info.manufacturer << "\"" << std::endl;
+
+    return true;
+}
+bool Task::startHook()
+{
+    if (! TaskBase::startHook())
+        return false;
+
+    mLastStatus = base::Time::now();
+    return true;
+}
+void Task::updateHook()
+{
+    TaskBase::updateHook();
+
+   if((base::Time::now() - mLastStatus) > _status_period.get())
+   {
+       mLastStatus = base::Time::now();
+       _acoustic_channel.write(getAcousticChannelparameters());
+   }
+
+
+}
+void Task::errorHook()
+{
+    TaskBase::errorHook();
+}
+void Task::stopHook()
+{
+    TaskBase::stopHook();
+}
+void Task::cleanupHook()
+{
+    TaskBase::cleanupHook();
+}
+void Task::processIO()
+{
+    AcousticConnection acoustic_connection = driver->getConnectionStatus();
+    _acoustic_connection.write(acoustic_connection);
+
+    // TODO define exactly what to do for each acoustic_connection status
+    if(acoustic_connection.status == ONLINE || acoustic_connection.status == INITIATION_ESTABLISH || acoustic_connection.status == INITIATION_LISTEN )
+    {
+        // Need a flow management of data be sent to device.
+        // Only send a new Instant Message if a acknowledgment notification of previously message was received if it was required.
+        while(IM_notification_ack && _message_input.read(send_IM) == RTT::NewData)
+        {
+            // Check free transmission buffer and instant message size.
+            checkFreeBuffer(driver->getStringOfIM(send_IM), acoustic_connection);
+            if(send_IM.buffer.size() > MAX_MSG_SIZE)
+                RTT::log(RTT::Error) << "Instant Message \""<< send_IM.buffer << "\" is longer than MAX_MSG_SIZE of \"" << MAX_MSG_SIZE << "\" bytes. It maybe not be sent to remote device " << std::endl;
+
+            driver->sendInstantMessage(send_IM);
+            if(send_IM.deliveryReport)
+                IM_notification_ack = false;
+        }
+
+        iodrivers_base::RawPacket raw_data_input;
+        while(_raw_data_input.read(raw_data_input) == RTT::NewData)
+        {
+            std::string buffer(raw_data_input.data.begin(), raw_data_input.data.end());
+            checkFreeBuffer(buffer, acoustic_connection);
+            if(driver->getMode() == DATA)
+                driver->sendRawData(raw_data_input.data);
+            else
+                RTT::log(RTT::Error) << "Device can not send raw_data \""<< raw_data_input.data.data() << "\" in COMMAND mode. Be sure to switch device to DATA mode" << std::endl;
```

Comment:
I think about create a property to establish mode (or a operation). So the user can choose with mode the device should operates. In DATA mode both commands and raw_data can be transmitted, in COMMAND mode only commands, including Instant Message. 
I'm note sure the best way to manage that internally, so I'd prefer the let the user decide it.